tutorial/markdown/nodejs/nodejs-langchain-pdf-chat/nodejs-langchain-pdf-chat.md

```diff
@@ -0,0 +1,678 @@
+                  "name": "embedding",
+                  "similarity": "dot_product",
+                  "type": "vector",
+                  "vector_index_optimized_for": "recall"
+                }
+              ]
+            },
+            "metadata": {
+              "dynamic": true,
+              "enabled": true
+            },
+            "text": {
+              "enabled": true,
+              "dynamic": false,
+              "fields": [
+                {
+                  "index": true,
+                  "name": "text",
+                  "store": true,
+                  "type": "text"
+                }
+              ]
+            }
+          }
+        }
+      }
+    },
+    "store": {
+      "indexType": "scorch",
+      "segmentVersion": 16
+    }
+  },
+  "sourceType": "gocbcore",
+  "sourceName": "pdf-docs",
+  "sourceParams": {},
+  "planParams": {
+    "maxPartitionsPerPIndex": 64,
+    "indexPartitions": 16,
+    "numReplicas": 0
+  }
+}
+```
+
+### Setup Environment Config
+
+Copy the `.env.template` file from the repo and rename it to `.env` (`.env.local` in case of local development) and replace the placeholders with the actual values for your environment.
+All configuration for communication with the database is read from the environment variables.
+
+```sh
+OPENAI_API_KEY=<open_ai_api_key>
+DB_CONN_STR=<connection_string_for_couchbase_cluster>
+DB_USERNAME=<username_for_couchbase_cluster>
+DB_PASSWORD=<password_for_couchbase_cluster>
+DB_BUCKET=<name_of_bucket_to_store_documents>
+DB_SCOPE=<name_of_scope_to_store_documents>
+DB_COLLECTION=<name_of_collection_to_store_documents>
+INDEX_NAME=<name_of_fts_index_with_vector_support>
+```
+
+> [Open AI](https://openai.com) API Key is required for usage in generating embedding and querying LLM
+
+> The [connection string](https://docs.couchbase.com/nodejs-sdk/current/howtos/managing-connections.html#connection-strings) expects the `couchbases://` or `couchbase://` part.
+
+> For this tutorial, `DB_BUCKET=pdf-chat`, `DB_SCOPE=_default`, `DB_COLLECTION=_default` and `INDEX_NAME=pdf_search`
+
+### Running the Application
+
+After starting couchbase server, adding vector index and installing dependencies. Our Application is ready to run.
+
+In the projects root directory, run the following command
+
+```sh
+npm run dev
+```
+
+The application will run on your local machine at (http://localhost:3000).
+
+### Using PDF Chat App
+
+At the home page, there will be option to upload the PDF. Drag and Drop the PDF file in the upload box or Click to browse. Click on Upload and the process of conversion starts. This step may take some time depending on the size of uploaded PDF.
+![Upload Screen View](upload_screen.png)
+
+After uploading, a new screen appears which shows the uploaded PDF at the left and chat interface on the right side. Use Chat to ask questions regarding the PDF and get answers.
+![Chat Screen View](chat_screen.png)
+
+## Concepts
+
```

Comment:
```suggestion

```

Reply:
I am using prettier, so prettier suggests to have this new line. I am also thinking of making using prettier mandatory for all future tutorials for consistent formatting